### PR TITLE
Filter out requests based on duration using Extensions

### PR DESCRIPTION
**Pull request by heyams**

### CODE REVIEW

File: opentelemetry-api/java-agent/TelemetryFilteredBaseOnRequestDuration/pom.xml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    <dependency>
+      <groupId>io.opentelemetry.javaagent</groupId>
+      <artifactId>opentelemetry-javaagent-extension-api</artifactId>
+      <version>2.1.0-alpha</version>
+    </dependency>
+    <dependency>
+      <groupId>com.microsoft.azure</groupId>
+      <artifactId>applicationinsights-agent</artifactId>
+      <version>${applicationinsights.agent.version}</version>
+    </dependency>
```
Comment (trask) [inline opentelemetry-api/java-agent/TelemetryFilteredBaseOnRequestDuration/pom.xml:56]:
```suggestion
```

File: opentelemetry-api/java-agent/extensions/CustomSpanExporter/src/main/java/com/example/DemoSpanExporter.java [hunk 2]
```diff
@@ -17,6 +17,7 @@ public class DemoSpanExporter implements SpanExporter {
 
     public final SpanExporter delegate;
 
+
```
Comment (trask) [inline opentelemetry-api/java-agent/extensions/CustomSpanExporter/src/main/java/com/example/DemoSpanExporter.java:20]:
```suggestion
```

File: opentelemetry-api/java-agent/pom.xml [hunk 3]
```diff
@@ -15,6 +15,8 @@
       <module>TrackCustomException</module>
       <module>TelemetryFilteredBaseOnSpanEvents</module>
       <module>extensions</module>
+        <module>TelemetryFilteredBaseOnRequestDuration</module>
+        <module>extensions/FilterSpanBasedOnDuration</module>
```
Comment (trask) [inline opentelemetry-api/java-agent/pom.xml:19]: (just aligning with above)
```suggestion
      <module>TelemetryFilteredBaseOnRequestDuration</module>
      <module>extensions/FilterSpanBasedOnDuration</module>
```

File: opentelemetry-api/java-agent/extensions/FilterSpanBasedOnDuration/src/main/java/com/example/DurationSpanExporter.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @Override
+    public CompletableResultCode export(Collection<SpanData> collection) {
+        List<SpanData> copiedList = new ArrayList<>();
+        for(SpanData spanData : collection) {
+            long duration = getRequestDurationInSeconds(spanData);
+            if (duration < 5) { // ignore spans with duration greater than 5 seconds
+                copiedList.add(spanData);
+            }
```
Comment (trask) [inline opentelemetry-api/java-agent/extensions/FilterSpanBasedOnDuration/src/main/java/com/example/DurationSpanExporter.java:27]: generally customers want to ignore fast spans and keep slow spans

File: opentelemetry-api/java-agent/extensions/FilterSpanBasedOnDuration/src/main/java/com/example/DurationSpanExporter.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        }
+        return delegate.export(copiedList);
+    }
+
+    private static long getRequestDurationInSeconds(SpanData spanData) {
```
Comment (trask) [inline opentelemetry-api/java-agent/extensions/FilterSpanBasedOnDuration/src/main/java/com/example/DurationSpanExporter.java:32]: I'd probably recommend using milliseconds, or converting to seconds as a double (so much truncation happens when converting seconds to long/int)

**Review by trask — APPROVED** [review]

File: opentelemetry-api/java-agent/extensions/FilterSpanBasedOnDuration/src/main/java/com/example/DurationSpanExporter.java (hunk 4 above)
Comment (heyams) [reply opentelemetry-api/java-agent/extensions/FilterSpanBasedOnDuration/src/main/java/com/example/DurationSpanExporter.java:27]: i will address it in a follow up pr
i set auto-merge.

**Pull request merged**